### PR TITLE
feat: Add resource group specific create functions

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Add specific Create functions for each resource group, to return specific type and props.

Signed-off-by: Darren Murray <darren.murray@lacework.net>

### CODE REVIEW

File: api/resource_groups_aws.go [hunk 1]
```diff
@@ -50,17 +50,31 @@ func (svc *ResourceGroupsService) UpdateAwsResourceGroup(data ResourceGroup) (
...
+func (svc *ResourceGroupsService) CreateAwsResourceGroup(data ResourceGroup) (
+	response AwsResourceGroupResponse,
+	err error,
+) {
+	err = svc.create(data, &response)
+	if err != nil {
+		return
+	}
+
+	return
```
Comment (afiune) [inline api/resource_groups_aws.go:73]:
```suggestion
	err = svc.create(data, &response)
	return
```

File: api/resource_groups_aws_test.go [hunk 2]
```diff
@@ -76,14 +76,14 @@ func TestResourceGroupsAwsUpdate(t *testing.T) {
 
 		if assert.NotNil(t, r.Body) {
 			body := httpBodySniffer(r)
-			assert.Contains(t, body, resourceGUID, "ResourceGUID missing")
+			fmt.Println(body)
```
Comment (afiune) [inline api/resource_groups_aws_test.go:79]: ?

File: api/resource_groups_azure.go [hunk 3]
```diff
@@ -50,17 +50,32 @@ func (svc *ResourceGroupsService) UpdateAzureResourceGroup(data ResourceGroup) (
...
+func (svc *ResourceGroupsService) CreateAzureResourceGroup(data ResourceGroup) (
+	response AzureResourceGroupResponse,
+	err error,
+) {
+	err = svc.create(data, &response)
+	if err != nil {
+		return
+	}
+
+	return
```
Comment (afiune) [inline api/resource_groups_azure.go:74]:
```suggestion
	return
```

File: api/resource_groups_machine_test.go [hunk 4]
```diff
@@ -95,19 +94,17 @@ func TestResourceGroupsMachineUpdate(t *testing.T) {
...
+			MachineTags: []map[string]string{{"tag": "machineTag"}},
 		},
 	)
 	assert.Equal(t, "group_name", resourceGroup.Name, "Machine Resource Group name mismatch")
 	assert.Equal(t, "MACHINE", resourceGroup.Type, "a new Machine Resource Group should match its type")
 	assert.Equal(t, 1, resourceGroup.Enabled, "a new Machine Resource Group should be enabled")
 	resourceGroup.ResourceGuid = resourceGUID
 
-	response, err := c.V2.ResourceGroups.UpdateMachineResourceGroup(resourceGroup)
+	response, err := c.V2.ResourceGroups.UpdateMachineResourceGroup(&resourceGroup)
```
Comment (afiune) [inline api/resource_groups_machine_test.go:107]: do you need to pass the pointer to this function?

Comment (dmurray-lacework) [reply api/resource_groups_machine_test.go:107]: Part of the workaround for invalid request when resourceGUID is included in the request is to strip the resourceGUID prior to sending the update. We can come up with a better workaround

```
func (group *ResourceGroupData) resetResourceGUID() {
	group.ResourceGuid = ""
}
```

File: api/resource_groups.go [hunk 5]
```diff
@@ -138,6 +139,25 @@ func (svc *ResourceGroupsService) Create(group ResourceGroupData) (
 	return
 }
 
+// Update updates a single ResourceGroup on the Lacework Server
+func (svc *ResourceGroupsService) Update(data ResourceGroup) (
+	response ResourceGroupResponse,
+	err error,
+) {
+	if data == nil {
+		return ResourceGroupResponse{}, errors.New("resource group must not be empty")
```
Comment (afiune) [inline api/resource_groups.go:148]:
```suggestion
		err = errors.New("resource group must not be empty")
		return
```

File: api/resource_groups_aws.go [hunk 6]
```diff
@@ -45,22 +45,34 @@ func (svc *ResourceGroupsService) GetAwsResourceGroup(guid string) (
...
-	response AwsResourceGroupResponse,
-	err error,
-) {
-	var rawResponse resourceGroupWorkaroundResponse
-	err = svc.update(data.ID(), data, &rawResponse)
+// UpdateAws updates a single Aws ResourceGroup on the Lacework Server
+func (svc *ResourceGroupsService) UpdateAws(data ResourceGroup) (
+	response AwsResourceGroupResponse, err error) {
+	if data == nil {
+		return AwsResourceGroupResponse{}, errors.New("resource group must not be empty")
```
Comment (afiune) [inline api/resource_groups_aws.go:52]:
```suggestion
		err = errors.New("resource group must not be empty")
		return
```

**Review by afiune — APPROVED** [review]
Just one nitpick that we agreed you will update but out of that, we are good to go!

tenor-151032375

**Pull request merged**